admin-dev/themes/new-theme/js/components/grid/extension/column/common/async-toggle-column-extension.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+const $ = window.$;
+
+/**
+ * Class AsyncToggleColumnExtension submits toggle action using AJAX
+ */
+export default class AsyncToggleColumnExtension {
+
+  constructor() {
+    return {
+      extend: (grid) => this.extend(grid),
+    }
+  }
+
+  /**
+   * Extend grid
+   *
+   * @param {Grid} grid
+   */
+  extend(grid) {
+    grid.getContainer().find('.js-grid-table').on('click', '.ps-togglable-row', (event) => {
+      event.preventDefault();
+
+      const $button = $(event.currentTarget);
+
+      $.post({
+        url: $button.data('toggle-url'),
+      }).then((response) => {
+        if (response.status) {
```

Comment:
What status can be returned? an http code? if yes, even if you get a 500 it will be a success.

Reply:
nope, it's 1 or 0, i think it was used by legacy that way, but i may be wrong. either way, i know its not the best implementation and since we dont have any rules for BO API, i dont want to invent anything new here. :/

Reply:
Then add a `@todo` here ;)